packages/grafana-ui/src/components/uPlot/plugins/TooltipPlugin.tsx

```diff
@@ -50,8 +50,10 @@ export const TooltipPlugin: React.FC<TooltipPluginProps> = ({
   const [coords, setCoords] = useState<CartesianCoords2D | null>(null);
   const [isActive, setIsActive] = useState<boolean>(false);
   const isMounted = useMountedState();
+  const search = new URLSearchParams(location.search);
 
   const pluginId = `TooltipPlugin`;
+  const isEditPanel = search.has('editPanel');
```

Comment:
It would be better to use  `usePanelContext` and check app == CoreApp.PanelEditor

Reply:
This is definitely a cleaner way of doing it. But the problem is, when you enter `Edit mode` on a graph panel, it switches to an Angular component that does not have the `PanelContext` on it, so it never updates and sets the `app` property to `CoreApp.PanelEditor` .

This function gives a clearer explanation: https://github.com/grafana/grafana/blob/main/public/app/features/dashboard/dashgrid/DashboardPanel.tsx#L83

The `PanelChrome` has `PanelContext`, but the `PanelChromeAngular` doesn't. And when we enter edit mode on a graph panel, we never update the context.

Another way to fix this (that I see) is to add another context across both `PanelChrome` and `PanelChromeAngular`  and also keep the context with `PanelChrome` since changing things there might break stuff in other places across Grafana. 

I'm not sure if this is the better approach. It will mean more complex code that might not be needed, especially since the plan is to deprecate Angular.

What do you think?

Reply:
@mdvictor ah I understand the issue now, the problem is that angular panels are emitting these evens even in panel edit  (react panels do not do this, because of this: https://github.com/grafana/grafana/blob/main/public/app/features/dashboard/dashgrid/PanelChrome.tsx#L83) .

For angular panels it's a bit different, but I would add a similar if condition here: https://github.com/grafana/grafana/blob/main/public/app/plugins/panel/graph/graph_tooltip.ts#L173  

some thing like: 

```ts
// broadcast to other graph panels that we are hovering!
if (!dashboard.panelInEdit) {
    pos.panelRelY = (pos.pageY - elem.offset().top) / elem.height();
    hoverEvent.payload.pos = pos;
    hoverEvent.payload.panel = panel;
    hoverEvent.payload.point['time'] = (pos as any).x;
    dashboard.events.publish(hoverEvent);
}
```

Reply:
Yes, you are right! This is the correct way of fixing it. 

Modified that bit of code, tested it and works like a charm!